runtime/lua/vim/lsp/completion.lua

```diff
@@ -284,6 +304,7 @@ function M._lsp_to_complete_items(result, prefix, client_id)
   local candidates = {}
   local bufnr = api.nvim_get_current_buf()
   local user_convert = vim.tbl_get(buf_handles, bufnr, 'convert')
+  local kind_highlights = generate_kind_hlgroup()
```

Comment:
Why generate this for every completion request? It only needs to be constructed once.

Reply:
highlight group can be added in runtime . i don't think this need generate once

Reply:
You can still use an autocommand for tracking when the highlight groups have changed right?

Reply:
idk what you mean. there still have group does not defined in default .  you can add a highlight group from ex highlight command .  how you tracking this when you do `:hi  Reference guifg=#fffffff`